helix-core/src/movement.rs

```diff
@@ -147,9 +163,14 @@ where
 /// Possible targets of a word motion
 #[derive(Copy, Clone, Debug)]
 pub enum WordMotionTarget {
+    ThisWordStart,
```

Comment:
I'd like to see what @PabloMansanet thinks, but I think we could avoid adding more enum types by simply using `skip_while`/`backwards_skip_while`. You could add a new function called `word_bounds` that would scan in both directions until it finds a different category.

Reply:
I don't have a particular complaint about expanding this enum (in fact, I expand it myself on #390) as long as the semantics of the enum are respected. Are these new "motion targets", or are we cramming unrelated concepts in the enum?

They seem fine to me. "Move until the start of this word" definitely has a different target than "Move to the start of the previous word" or "move to the end of the previous word".

The only one I'm iffy about is `ThisWordPrevBound`, because it doesn't seem to have a different target than `PrevWordEnd`, just different conditions that could maybe be expressed differently. For example, the entry for `PrevWordEnd` could become `PrevWordEnd { sticky: bool }`

Reply:
I converted `ThisWordPrevBound` to `PrevWordEnd { sticky: true }`, works better too 👍🏾